compiler/test/fail_compilation/diag23355.d

```diff
@@ -0,0 +1,11 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/diag23355.d(9): Error: undefined identifier `n`
+fail_compilation/diag23355.d(11): Error: none of the overloads of template `diag23355.ffi` are callable using argument types `!()(int[4])`
+fail_compilation/diag23355.d(9):        Candidate is: `ffi(T)(T[n] s)`
+---
+*/
+void ffi(T)(T[n] s) { }
+
```

Comment:
0 template parameter test
```suggestion
void ffi()(T[n] s)
{
}
```

Reply:
that one is unrelated to this fix, already worked before

Reply:
this is in reference to your point about parameters being of zero length.

Reply:
ok added and using inferStart now

Reply:
![image](https://user-images.githubusercontent.com/5672948/191619169-449181c0-c956-4d55-a55d-95ade91ab398.png)
gib coauthor

Reply:
done :p